SubmittingPatches-backports.rst

```diff
@@ -234,12 +234,26 @@ the backport branch with the cherry-picks in it.
 
 The script is located at ``src/script/ceph-backport.sh`` in the ``master``
 branch. Though there might be an older version of this script in a stable
-branch, do not use it. Only use the most recent version from master.
+branch, do not use it. Only use the most recent version from master and
+best not from your active repository. That said, copy the script to a
+dedicated path outside the repository you use or use the following
+alias that will download the most recent script on every call::
+
+    alias ceph-backport="bash <(curl -Ls https://github.com/ceph/ceph/raw/master/src/script/ceph-backport.sh)"
```

Comment:
Ugh. I admit I'm not very fond of this approach, curling a script into bash is usually considered very bad practice, especially for a script that you intent to use several times and not just once.

Reply:
I just copy the script into `~/bin`.

Reply:
Thanks for the tip, used it. But used the `upstream/master` instead, as you don't have to keep your local master in sync if you create you branches with `git checkout -b $branch upstream/master`. Removed all other suggestions as it looks like a perfect fit.

Reply:
Well, this command is not guaranteed to give you what is in the latest upstream master. The first problem is that the user's "upstream" remote might not be called "upstream" (mine is called "ceph", other folks call it "origin", etc.). The second problem is that "upstream/master" is still just a local snapshot. You'd have to do "git fetch upstream", first, to ensure it is up-to-date.

All that being said, I think this is a nice trick and folks who are interested in such things will be able to make it work.

Reply:
Sure my upstream is called ceph, too. But you need also to put the right path to your clone in it. In order to backport you also have to fetch the latest upstream changes. As this documentation is for developers who should know at least where their clone is and how to update it. I think it's a good approach to say that this is the best way without wasting a lot of time (in sum) to download the script for every use.